### ISSUE TITLE
fresh kubeadm 1.5 install fails due to existing /var/lib/kubelet

### ISSUE THREAD

**Issue by ozdanborne** [kubernetes/kubeadm#262]
If you follow the kubeadm 1.5 install instructions to the T your master won't succesfully init because `/var/lib/kubelet` already exists. seems `/var/lib/kubelet` is created when installing kubelet in this line

```
sudo apt-get --y install kubectl=1.5.3-00 kubelet=1.5.3-00 kubernetes-cni=0.3.0.1-07a8a2-00
```

So users will need to `kubeadm reset` before their first  `kubeadm init`.

**Comment by kartoch** [comment]
Problem still present in 1.6.3.

**Comment by luxas** [comment]
@dgoodwin Did you ever find out why this race condition exists?

@timothysc Have you seen/looked into this?

**Comment by dgoodwin** [comment]
I did not but there is some analysis in https://github.com/kubernetes/release/issues/171#issuecomment-270334974 and an open PR in https://github.com/kubernetes/release/pull/256

**Comment by ianvernon** [comment]
I'm hitting this in the 1.6.4 version of kubeadm as well.

**Comment by madorn** [comment]
same here.

**Comment by luxas** [comment]
Can any of you reproduce this with kubeadm v1.7?

**Comment by jstomphorst** [comment]
I repoduced it with a centos 7 and kubeadm v1.8 but kubeadm reset helps...

**Comment by jamiehannaford** [comment]
https://github.com/kubernetes/kubernetes/pull/53317 should also help with this (since the kubelet was saving its PKI assets in `/var/run/kubelet`). Hopefully when we have that in kubeadm, we can close this issue after a suitable waiting period (unless anybody runs into related issues).

**Comment by seslav** [comment]
I'm having the same issue and cannot start kubeadm 

kubectl-1.8.0-0.x86_64
kubelet-1.8.0-0.x86_64
kubeadm-1.8.0-1.x86_64
kubernetes-cni-0.5.1-0.x86_64

Any workarounds?

**Comment by seslav** [comment]
Figure it out.

So if you see the following

> [init] Using Kubernetes version: v1.8.0
[init] Using Authorization modes: [Node RBAC]
[preflight] Running pre-flight checks
[preflight] WARNING: kubelet service is not enabled, please run 'systemctl enable kubelet.service'
[preflight] WARNING: firewalld is active, please ensure ports [6443 10250] are open or your cluster may not function correctly
[preflight] WARNING: Running with swap on is not supported. Please disable swap or set kubelet's --fail-swap-on flag to false.
[preflight] Some fatal errors occurred:
	/var/lib/kubelet is not empty
[preflight] If you know what you are doing, you can skip pre-flight checks with `--skip-preflight-checks`

> 

execute **kubeadm reset**  then **kubeadm init** That should do the work

**Comment by seslav** [comment]
I found a solution for me. The problem was the swap
So I executed **swapoff -a** then **kubeadm reset** and finally **kubeadm init**

**Comment by luxas** [comment]
v1.5 isn't supported anymore. v1.8.0 had a problem with similar symptoms, but it was fixed in v1.8.1
